cgen error when use reference with fixed array `_ := &[2]u8{}`

### Describe the bug

cgen error

### Reproduction Steps

`_ := &[2]u8{}`

### Expected Behavior

no cgen error

### Current Behavior

```bash
==================
/tmp/v_1000/discardable.2943580652518909944.tmp.c: In function ‘main__main’:
/tmp/v_1000/discardable.2943580652518909944.tmp.c:12695:61: error: macro "HEAP" passed 3 arguments, but takes just 2
12695 |         {Array_fixed_u8_2* _ = HEAP(Array_fixed_u8_2, {0, 0});}
      |                                                             ^
/tmp/v_1000/discardable.2943580652518909944.tmp.c:187: note: macro "HEAP" defined here
  187 | #define HEAP(type, expr) ((type*)memdup((void*)&((type[]){expr}[0]), sizeof(type)))
      | 
/tmp/v_1000/discardable.2943580652518909944.tmp.c:12695:32: error: ‘HEAP’ undeclared (first use in this function)
12695 |         {Array_fixed_u8_2* _ = HEAP(Array_fixed_u8_2, {0, 0});}
      |                                ^~~~
/tmp/v_1000/discardable.2943580652518909944.tmp.c:12695:32: note: each undeclared identifier is reported only once for each function it appears in
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.3 cb28144

### Environment details (OS name and version, etc.)

Ubuntu
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -24,7 +24,7 @@ fn (mut g Gen) array_init(node ast.ArrayInit, var_name string) {
 	}
 	len := node.exprs.len
 	if array_type.unaliased_sym.kind == .array_fixed {
-		g.fixed_array_init(node, array_type, var_name)
+		g.fixed_array_init(node, array_type, var_name, is_amp)
 		if is_amp {
 			g.write(')')
 		}
@@ -79,7 +79,7 @@ fn (mut g Gen) array_init(node ast.ArrayInit, var_name string) {
 	}
 }
 
-fn (mut g Gen) fixed_array_init(node ast.ArrayInit, array_type Type, var_name string) {
+fn (mut g Gen) fixed_array_init(node ast.ArrayInit, array_type Type, var_name string, is_amp bool) {
 	prev_inside_lambda := g.inside_lambda
 	g.inside_lambda = true
 	defer {
@@ -177,6 +177,8 @@ fn (mut g Gen) fixed_array_init(node ast.ArrayInit, array_type Type, var_name st
 				g.write(', ')
 			}
 		}
+	} else if is_amp {
+		g.write('0')
 	} else {
 		elem_sym := g.table.final_sym(node.elem_type)
 		if elem_sym.kind == .map {

--- a/vlib/v/gen/c/dumpexpr.v
+++ b/vlib/v/gen/c/dumpexpr.v
@@ -210,8 +210,13 @@ fn (mut g Gen) dump_expr_definitions() {
 		surrounder.builder_write_afters(mut dump_fns)
 		if is_fixed_arr_ret {
 			tmp_var := g.new_tmp_var()
-			dump_fns.writeln('\t${str_dumparg_ret_type} ${tmp_var} = {0};')
-			dump_fns.writeln('\tmemcpy(${tmp_var}.ret_arr, dump_arg, sizeof(${str_dumparg_type}));')
+			if typ.is_ptr() {
+				dump_fns.writeln('\t${str_dumparg_ret_type} ${tmp_var} = HEAP(${g.typ(typ.set_nr_muls(0))}, {0});')
+				dump_fns.writeln('\tmemcpy(${tmp_var}->ret_arr, dump_arg, sizeof(${str_dumparg_type}));')
+			} else {
+				dump_fns.writeln('\t${str_dumparg_ret_type} ${tmp_var} = {0};')
+				dump_fns.writeln('\tmemcpy(${tmp_var}.ret_arr, dump_arg, sizeof(${str_dumparg_type}));')
+			}
 			dump_fns.writeln('\treturn ${tmp_var};')
 		} else {
 			dump_fns.writeln('\treturn dump_arg;')

--- a/vlib/v/tests/array_fixed_ptr_test.v
+++ b/vlib/v/tests/array_fixed_ptr_test.v
@@ -1,0 +1,11 @@
+fn test_main() {
+	a := &[2]u8{}
+	dump(a)
+	assert a.len == 2
+}
+
+fn test_empty() {
+	_ := &[2]u8{}
+
+	assert true
+}

